crml/cennzx/rpc/src/lib.rs

```diff
@@ -66,6 +80,122 @@ impl<C, T> Cennzx<C, T> {
+pub struct SellPriceResponse<Balance> {
+	#[serde(with = "serde_balance")]
+	price: Balance,
+}
+
+#[derive(Eq, PartialEq, Decode, Encode, Default, Debug, Serialize, Deserialize)]
+#[serde(bound(serialize = "Balance: std::fmt::Display"))]
+#[serde(bound(deserialize = "Balance: std::str::FromStr"))]
+pub struct LiquidityValueResponse<Balance> {
+	#[serde(with = "serde_balance")]
+	liquidity: Balance,
+
+	#[serde(with = "serde_balance")]
+	core: Balance,
+
+	#[serde(with = "serde_balance")]
+	asset: Balance,
+}
+
+#[derive(Eq, PartialEq, Decode, Encode, Default, Debug, Serialize, Deserialize)]
+#[serde(bound(serialize = "Balance: std::fmt::Display"))]
+#[serde(bound(deserialize = "Balance: std::str::FromStr"))]
+pub struct LiquidityPriceResponse<Balance> {
+	#[serde(with = "serde_balance")]
+	core: Balance,
+
+	#[serde(with = "serde_balance")]
+	asset: Balance,
+}
+
+mod serde_balance {
+	use serde::{Deserialize, Deserializer, Serializer};
+
+	pub fn serialize<S: Serializer, Balance: std::fmt::Display>(t: &Balance, serializer: S) -> Result<S::Ok, S::Error> {
+		serializer.serialize_str(&t.to_string())
+	}
+
+	pub fn deserialize<'de, D: Deserializer<'de>, Balance: std::str::FromStr>(
+		deserializer: D,
+	) -> Result<Balance, D::Error> {
+		let s = String::deserialize(deserializer)?;
+		s.parse::<Balance>()
+			.map_err(|_| serde::de::Error::custom("Parse from string failed"))
+	}
+}
+
+// A balance type for receiving over RPC
+pub struct WrappedBalance(u128);
+#[derive(Debug, Default, Serialize, Deserialize)]
+/// Private, used to help serde handle `WrappedBalance`
+/// https://github.com/serde-rs/serde/issues/751#issuecomment-277580700
+struct WrappedBalanceHelper {
+	value: u128,
+}
+impl Serialize for WrappedBalance {
+	fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		WrappedBalanceHelper { value: self.0 }.serialize(serializer)
+	}
+}
+
+impl<'de> Deserialize<'de> for WrappedBalance {
+	fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+	where
+		D: Deserializer<'de>,
+	{
+		deserializer.deserialize_any(WrappedBalanceVisitor).map_err(|err| {
+			println!("{:?}", err);
+			serde::de::Error::custom("deserialize failed")
+		})
+		// let value = TryFrom::try_from(u256).map_err(|_| serde::de::Error::custom("Try from U256 failed"))?;
+		// Ok(WrappedBalance(1_u128))
+	}
+}
+
+struct WrappedBalanceVisitor;
+
+impl<'de> serde::de::Visitor<'de> for WrappedBalanceVisitor {
+	type Value = WrappedBalance;
+	fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+		write!(formatter, "an integer or hex-string")
+	}
+
+	fn visit_u64<E>(self, v: u64) -> std::result::Result<Self::Value, E>
+	where
+		E: serde::de::Error,
+	{
+		Ok(WrappedBalance(v.saturated_into()))
+	}
+
+	fn visit_str<E>(self, s: &str) -> std::result::Result<Self::Value, E>
+	where
+		E: serde::de::Error,
+	{
+		let decoded_string =
```

Comment:
could you add a comment that we're expecting a `0x` prefixed hex string which is why we have the `2..`